docker-compose.yml

```diff
@@ -429,10 +415,126 @@ services:
       - 127.0.0.1:8093:8080
     networks:
       placeos:
-        ipv4_address: 172.31.231.19
     volumes:
       - type: volume
         source: rethink-data
         target: /data/rethinkdb_data
     environment:
       TZ: $TZ
+
+  # Aggregates logs and forwards them to Elasticsearch.
+  logstash:
+    image: docker.elastic.co/logstash/logstash-oss:${ELASTIC_VERSION:-7.6}
+    restart: always
+    container_name: logstash
+    hostname: logstash
+    expose:
+      - 12201/udp
+      - 5044
+    ports:
+      - "127.0.0.1:12201:12201/udp"
+    networks:
+      placeos:
+    volumes:
+      - ./config/logstash/config:/config
+      - ./config/logstash/patterns:/opt/logstash/extra_patterns
+    restart: always
+    labels:
+      container_group: logging
+    logging:
+      driver: json-file
+      options:
+        max-size: 99m
+        labels: container_group
+    command: logstash -f /config
+
+  # Run 'docker-compose run --rm validate-logstash-config' to quickly check the logstash config.
+  validate-logstash-config:
+    container_name: validate-logstash
+    image: docker.elastic.co/logstash/logstash-oss:${ELASTIC_VERSION:-7.6}
+    volumes:
+      - ./config/logstash/config:/config
+    command: logstash -t -f /config
+
+  # Sends all container json-file logs to logstash
+  logspout:
+    image: vincit/logspout-gelf
+    hostname: $MONITOR_HOSTNAME
+    container_name: "logspout"
+    networks:
+      placeos:
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+    command: gelf://$LOGSTASH_HOST:$LOGSTASH_PORT
+    restart: unless-stopped
+    logging:
+      driver: json-file
+      options:
+        max-size: 99m
+
+  # Gets metrics from host machine and send to elastic
+  metricbeat:
+    image: docker.elastic.co/beats/metricbeat-oss:${ELASTIC_VERSION:-7.6}
+    hostname: $MONITOR_HOSTNAME
+    container_name: "metricbeat"
+    user: root
+    networks:
+      placeos:
+    volumes:
+      - /proc:/hostfs/proc:ro
+      - /sys/fs/cgroup:/hostfs/sys/fs/cgroup:ro
+      - /:/hostfs:ro
+      - /var/run/docker.sock:/var/run/docker.sock:ro
+      - ./config/metricbeat.yml:/usr/share/metricbeat/metricbeat.yml
```

Comment:
Also not guaranteed to work for macos

Reply:
If we're going down this route, we absolutely need to start templating out our docker-compose.

@kimburgess for our reworked environment, as part of the install it would be nice to template out the env correctly for the host. the same tooling can be leveraged for changing the container pins as well.

Reply:
It looks like /var/run/docker.sock will work in macos - https://github.com/google/cadvisor/issues/1565#issuecomment-718812180

Though the metricbeat docs do say /proc and /sys are only available on linux and don't mention a workaround for mac
```
The special filesystems /proc and /sys are only available if the host system is running Linux. Attempts to bind-mount these filesystems will fail on Windows and MacOS.
```
https://www.elastic.co/guide/en/beats/metricbeat/current/running-on-docker.html#monitoring-host

Reply:
We'll need to template that at a minimum it seems

Reply:
@caspiano When you have some time could you please double check the latest of this branch on mac?
https://github.com/place-labs/partner-environment/pull/34/commits/4f1e6167abd798f6c960c130f76ad2f1aea5852e

Looking for metricbeat to stay up and not fail due to volume issues. It should not set the env vars and use /dev/null for the /proc and /sys/fs/cgroup paths, and will mount /var/run/docker.sock

Reply:
getting this (added some newlines for readability)

```
ERROR: for metricbeat  
Cannot start service metricbeat: 
OCI runtime create failed:container_linux.go:
367: starting container process caused: process_linux.go:
495: container init caused: rootfs_linux.go:
60: mounting "/dev/null" to rootfs at "/var/lib/docker/overlay2/81152ea54bfb75f778b4aaf4e11f20c3657d1fe8abc9aa7c0d5b9deafb6becca/merged/hostfs/proc" caused: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type
ERROR: Encountered errors while bringing up the project.
```

Reply:
metricbeat removed from mac deploy https://github.com/place-labs/partner-environment/pull/34/commits/0b86dcba300ad4976c050cb17cdaa3a330d7062f